visualizer/src/bin/etch_a_sketch.cpp

```diff
@@ -111,26 +114,29 @@ struct EtchaSketchPlayer : public audio::Player<EtchaSketchPlayer> {
                     self->current_pot_node = self->serial_reciever.pot_vals_head;
                 }
                 /* Increment the trace time */
-                self->current_trace_time += (1.0 / self->serial_reciever.pot_num_values) * (1.0 / self->serial_reciever.trace_freq);
+                accurate_float trace_contribution = (accurate_float(1) / (accurate_float(self->serial_reciever.pot_num_values) * accurate_float(self->serial_reciever.trace_freq)));
+                self->current_trace_time += trace_contribution;
```

Comment:
implicit casting bites again.. turns out there's more than enough precision in long doubles 

I get this locally 
```c++
eps examination
float eps 1.19209289550781250000e-07
long double eps 1.08420217248550443401e-19
```

this line gives you zero because the lhs and rhs get cast to floats and then implicitly cast the result to a long double which destroys the precision 
```c++
self->current_trace_time += (1.0 / self->serial_reciever.pot_num_values) * (1.0 / self->serial_reciever.trace_freq);
```